docs/docs/administration/single-sign-on.md

```diff
@@ -0,0 +1,218 @@
+
+You can extend the providers in your `sso.yml` file with additional settings to restrict access and define default roles for users. These fields are optional, but **highly recommended** for tightening security and streamlining user provisioning.
+
+### Restrict access to specific email domains
+
+Use the `allowed_email_domains` setting to restrict SSO logins to specific email domains. This is useful if your identity provider manages multiple domains or if you want to prevent unauthorized domains from accessing your Mathesar instance.
+
+- **Default:** No restriction (i.e., users from any domain can log in).
+- **Expected format:** A list of domain names (e.g., `['example.com', 'mathesar.org']`).
+
+**Example:**
+
+```diff
+oidc_providers:
+  provider1:
+    provider_name: okta
+    server_url: https://trial-example-admin.okta.com
+    client_id: YOUR_CLIENT_ID
+    secret: YOUR_SECRET
++   allowed_email_domains: ['example.com', 'mathesar.org']
+```
+
+With this configuration, only users whose email ends in `@example.com` or `@mathesar.org` will be allowed to log in.
+
+---
+
+### Set default user roles for your databases
+
+The `default_pg_role` block allows you to **automatically assign PostgreSQL roles** to users the first time they log in via SSO. You can configure this for one or more databases.
+
+Each database block must include the following:
+
+- `name`: The **PostgreSQL database name**
+- `host`: The **hostname** or IP address of the database
+- `port`: The **port** on which the database is running
+- `role`: The **PostgreSQL role** to assign to users
+
+**Example:**
+
+```diff
+default_pg_role:
++ db1:
++   name: my_database
++   host: db.internal.example.com
++   port: 5432
++   role: readonly_user
++ db2:
++   name: analytics_db
++   host: analytics-db.example.net
++   port: 5432
++   role: analyst
+```
+
+On first login, users will be granted the specified roles on each listed database. This simplifies onboarding and ensures consistent access control across your environment.
+
+## Disabling Single Sign-On
+
+To disable SSO in Mathesar, delete the `sso.yml` file from your installation directory and restart the application. This will revert Mathesar to using only email and password-based authentication.
+
+!!!warning "Resetting user passwords"
+    Users who were originally created via SSO do not have a known password—they are automatically assigned a random, system-generated one during account creation. As a result, these users will not be able to log in after SSO is disabled unless their passwords are reset.
+
+    To restore access for these users:
+
+    1. Have an administrator reset each user’s password through the Mathesar user interface.
+    2. When the user logs in with the administrator-supplied password, they will be prompted to create a new personal password.
+
+    We recommend completing all necessary password resets before disabling SSO to ensure a smooth transition and avoid user lockouts.
```

Comment:
@Anish9901 is this accurate?

Reply:
Yeah, mostly accurate, if the users were initially created by an admin and a new password _is_ already created for a user, using SSO won't affect their account. They could login via traditional method and via SSO.